client-v2/src/test/java/com/clickhouse/client/query/QueryTests.java

```diff
@@ -65,65 +85,368 @@ public void testSimpleSelectTableFormat() {
+        Future<QueryResponse> response = client.query("SELECT * FROM " + table, null, settings);
+        TableSchema schema = client.getTableSchema(table);
+
+        QueryResponse queryResponse = response.get();
+        ClickHouseBinaryFormatReader reader = createBinaryFormatReader(queryResponse, settings, schema);
+
+        Assert.assertTrue(reader.next());
+        Map<String, Object> record = new HashMap<>();
+        reader.copyRecord(record);
+//        System.out.println("col1: " + Arrays.toString(col1Values));
+        System.out.println("Record: " + record);
+    }
+
+
+    @Test(groups = {"integration"})
+    public void testQueryExceptionHandling() {
+
+    }
+
+    private final static List<String> DATASET_COLUMNS = Arrays.asList(
+            "col1 UInt32",
+            "col2 Int32",
+            "col3 String",
+            "col4 Int64",
+            "col5 String"
+    );
+
+    private final static List<Function<String, Object>> DATASET_VALUE_GENERATORS = Arrays.asList(
+            c -> Long.valueOf(RANDOM.nextInt(Integer.MAX_VALUE)),
+            c -> RANDOM.nextInt(Integer.MAX_VALUE),
+            c -> "value_" + RANDOM.nextInt(Integer.MAX_VALUE),
+            c -> Long.valueOf(RANDOM.nextInt(Integer.MAX_VALUE)),
+            c -> "value_" + RANDOM.nextInt(Integer.MAX_VALUE)
+    );
+
+    private final static String DATASET_TABLE = "query_test_table";
+
+    private void prepareSimpleDataSet() {
+        prepareDataSet(DATASET_TABLE, DATASET_COLUMNS, DATASET_VALUE_GENERATORS, 1);
+    }
+
+    private List<Map<String, Object>> prepareDataSet(String table, List<String> columns, List<Function<String, Object>> valueGenerators,
+                                                     int rows) {
+        List<Map<String, Object>> data = new ArrayList<>(rows);
+        try (
+                ClickHouseClient client = ClickHouseClient.builder().config(new ClickHouseConfig())
+                        .nodeSelector(ClickHouseNodeSelector.of(ClickHouseProtocol.HTTP))
+                        .build()) {
+            // Drop table
+            ClickHouseRequest<?> request = client.read(getServer(ClickHouseProtocol.HTTP))
+                    .query("DROP TABLE IF EXISTS default." + table);
             request.executeAndWait();
 
-            request = client.write(getServer(ClickHouseProtocol.HTTP))
-                    .query("INSERT INTO default.query_test_table VALUES (1, 2), (3, 4), (5, 6)");
+
+
+            // Create table
+            StringBuilder createStmtBuilder = new StringBuilder();
+            createStmtBuilder.append("CREATE TABLE IF NOT EXISTS default.").append(table).append(" (");
+            for (String column : columns) {
+                createStmtBuilder.append(column).append(", ");
+            }
+            createStmtBuilder.setLength(createStmtBuilder.length() - 2);
+            createStmtBuilder.append(") ENGINE = Memory");

```

Comment:
Lets use mergetree and not memory mergetree is the main engine in ClickHouse i want things will be persist on disk and when we test cloud also want to make sure that we are testing mergetree

Reply:
ok. Will change.